src/xo-app/vm/tab-advanced.js

```diff
@@ -253,6 +256,39 @@ class CoresPerSocket extends Component {
   }
 }
 
+@addSubscriptions({
+  resourceSets: subscribeResourceSets,
+})
+class VmResourceSet extends Component {
+  _onChange = resourceSet =>
+    editVm(this.props.vm, { resourceSet: resolveId(resourceSet) })
```

Comment:
If you know `resourceSet` is an object, why not using `resourceSet.id` directly?